--- a/src/gn/rust_project_writer.cc
+++ b/src/gn/rust_project_writer.cc
@@ -236,8 +236,13 @@ void AddTarget(const BuildSettings* buil
   std::ranges::sort(include_dirs, [](const auto& lhs, const auto& rhs) {
     return lhs.value() < rhs.value();
   });
+#if defined(_GLIBCXX_RELEASE) && _GLIBCXX_RELEASE < 13
+  include_dirs.erase(std::unique(include_dirs.begin(), include_dirs.end()),
+                     include_dirs.end());
+#else
   include_dirs.erase(std::ranges::unique(include_dirs).begin(),
                      include_dirs.end());
+#endif
 
   Crate crate = Crate(crate_root, include_dirs, crate_id, crate_label,
                       edition.value_or("2015"));
--- a/src/gn/ninja_binary_target_writer.cc
+++ b/src/gn/ninja_binary_target_writer.cc
@@ -152,8 +152,13 @@ void NinjaBinaryTargetWriter::WritePriva
   std::ranges::sort(deps, [](const Target* lhs, const Target* rhs) {
     return lhs->module_name() < rhs->module_name();
   });
+#if defined(_GLIBCXX_RELEASE) && _GLIBCXX_RELEASE < 13
+  auto dirty = std::unique(deps.begin(), deps.end());
+  deps.erase(dirty, deps.end());
+#else
   auto dirty = std::ranges::unique(deps);
   deps.erase(dirty.begin(), dirty.end());
+#endif
   WriteModuleDeps(out, deps, base);
   out << "}\n";
 }
