--- mrtg-2.17.10.orig/Makefile.in
+++ mrtg-2.17.10/Makefile.in
@@ -37,7 +37,7 @@ bin/rateup.o: $(top_srcdir)/src/rateup.c
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c $(top_srcdir)/src/rateup.c -o bin/rateup.o
 
 subst:
-	$(PERL) -0777 -p -i~ -e "s'^#!\s*/\S*perl'#! $(PERL)'" $(top_srcdir)/bin/cfgmaker $(top_srcdir)/bin/indexmaker $(top_srcdir)/bin/mrtg
+	$(PERL) -0777 -p -i~ -e "s'^#!\s*/\S*perl'#! /usr/bin/perl'" $(top_srcdir)/bin/cfgmaker $(top_srcdir)/bin/indexmaker $(top_srcdir)/bin/mrtg
 	$(PERL) -0777 -p -i~ -e 's@GRAPHFMT="...";@GRAPHFMT="$(GDFORM_EXT)";@' $(top_srcdir)/bin/mrtg $(top_srcdir)/bin/indexmaker
 
 clean:
