--- mrtg-2.17.10.orig/configure.ac
+++ mrtg-2.17.10/configure.ac
@@ -20,46 +20,6 @@ fi
 
 AC_PATH_PROGS(NROFF, groff nroff)
 
-dnl Check for long long type (64 bit rateup)
-AC_CHECK_HEADERS(inttypes.h)
-AC_CHECK_TYPE(unsigned long long)
-AC_CHECK_TYPE(long long)
-AC_CHECK_FUNCS(strtoll)
-
-# Taken from libIDL-0.8.2
-AC_MSG_CHECKING([for printf long long format specifier])
-AC_CACHE_VAL(mr_cv_long_long_format_specifier,[
-    for format in ll l q I64; do
-        AC_RUN_IFELSE([AC_LANG_SOURCE([[#include <stdio.h>  
-            int main()
-            {
-                long long b, a = -0x3AFAFAFAFAFAFAFALL;
-                char buffer[1000];
-                sprintf (buffer, "%${format}u", a);
-                sscanf (buffer, "%${format}u", &b);
-                exit (b!=a);
-            }
-            ]])],[mr_cv_long_long_format_specifier="%${format}d"
-            mr_cv_long_long_format="${format}d"
-            break],[],[])
-        done])
-AC_MSG_RESULT($mr_cv_long_long_format_specifier)
-AC_DEFINE_UNQUOTED(LLD, "$mr_cv_long_long_format_specifier",
-    [How to print a long long])
-AC_DEFINE_UNQUOTED(LLD_FORMAT, "$mr_cv_long_long_format",
-    [long long format without % specifier])
-
-AH_BOTTOM(
-#ifndef HAVE_STRTOLL     
-long long int strtoll(const char *str, char **ptr, int base)
-{
-        long long int ll;
-        sscanf(str, LLD, &ll);
-        return ll;
-}
-#endif
-)
-
 dnl Checks for libraries.
 AC_CHECK_LIB(m, pow, [ MATHLIBS="-lm" ])
 
