diff --git a/dns_proxy.c b/dns_proxy.c
index 87083bf..519b223 100644
--- a/dns_proxy.c
+++ b/dns_proxy.c
@@ -219,10 +219,14 @@ int udp_listener() {
       error("[!] Error opening logfile.");
   }
 
-  if(geteuid() == 0) {
-    setuid(getpwnam(USERNAME)->pw_uid);
-    setgid(getgrnam(GROUPNAME)->gr_gid);
-  }
+  printf("[*] No errors, backgrounding process.\n");
+
+  // daemonize the process.
+  if(fork() != 0) { exit(0); }
+  if(fork() != 0) { exit(0); }
+
+  setuid(getpwnam(USERNAME)->pw_uid);
+  setgid(getgrnam(GROUPNAME)->gr_gid);
   socklen_t dns_client_size = sizeof(struct sockaddr_in);
 
   // setup SIGCHLD handler to kill off zombie children
