--- a/meson.build
+++ b/meson.build
@@ -64,15 +63,3 @@ pkg.generate(
   libraries: libjose,
   requires: 'jansson',
 )
-
-if a2x.found()
-  foreach m : mans
-    custom_target(m.split('/')[-1], input: m + '.adoc', output: m.split('/')[-1],
-      command: [a2x, '-f', 'manpage', '-D', meson.current_build_dir(), '@INPUT@'],
-      install_dir: join_paths(get_option('mandir'), 'man' + m.split('.')[-1]),
-      install: true
-    )
-  endforeach
-else
-  warning('Will not build man pages due to missing dependencies!')
-endif
