Index: accel-ppp-1.12.0/CMakeLists.txt
===================================================================
--- accel-ppp-1.12.0.orig/CMakeLists.txt
+++ accel-ppp-1.12.0/CMakeLists.txt
@@ -2,10 +2,6 @@ cmake_minimum_required(VERSION 2.6)
 cmake_policy(SET CMP0005 NEW)
 cmake_policy(SET CMP0003 OLD)
 
-if (CMAKE_BINARY_DIR STREQUAL CMAKE_HOME_DIRECTORY)
-    message(FATAL_ERROR "Building in source directory is forbidden. Please make separated build directory.")
-endif ()
-
 project(accel-ppp C)
 
 include(cmake/cpack.cmake)
Index: accel-ppp-1.12.0/accel-pppd/CMakeLists.txt
===================================================================
--- accel-ppp-1.12.0.orig/accel-pppd/CMakeLists.txt
+++ accel-ppp-1.12.0/accel-pppd/CMakeLists.txt
@@ -125,10 +125,3 @@ INSTALL(TARGETS accel-pppd
 
 INSTALL(FILES accel-ppp.conf.5 DESTINATION share/man/man5)
 
-IF (NOT DEFINED CPACK_TYPE)
-	INSTALL(FILES accel-ppp.conf DESTINATION ${CMAKE_FIND_ROOT_PATH}/etc RENAME accel-ppp.conf.dist)
-
-	INSTALL(CODE "EXECUTE_PROCESS(COMMAND mkdir -p ${CMAKE_FIND_ROOT_PATH}/var/log/accel-ppp)")
-	INSTALL(CODE "EXECUTE_PROCESS(COMMAND mkdir -p ${CMAKE_FIND_ROOT_PATH}/var/lib/accel-ppp)")
-ENDIF (NOT DEFINED CPACK_TYPE)
-
